kafka-cron/README.md

```diff
@@ -0,0 +1,162 @@
+
+Read the [Overview of Prometheus](https://prometheus.io/docs/introduction/overview/) if you are not familiar with Prometheus.
+
+The [Prometheus Guide to Instrumenting a Go Application](https://prometheus.io/docs/guides/go-application/) describes how to add metrics to a Golang application.
+
+First, consider:
+ * What kinds of things may go wrong with our system? (it is useful to look at errors your code is handling, as inspiration)
+ * What would users' expectations be of this system?
+ * What metrics can we add that will tell us when the system is not working as intended?
+ * What metrics can we add that might help us to troubleshoot the system and understand how it is operating? Read back through the first three parts of this exercise to try and identify the properties of the system that we might want to know about.
+
+Asking these questions should guide us in designing the metrics that our consumers and producer should export.
+Think about what kinds of problems can happen both in the infrastructure - Kafka, your consumers and producers - and in the submitted jobs. 
+
+Add metrics to your programs. Verify that they work as expected using `curl` or your web browser.
+
+Kafka doesn't export Prometheus metrics natively. However, we can use the official 
+[Prometheus JMX exporter](https://github.com/prometheus/jmx_exporter) to expose its metrics.
+Set this up (it is probably best as another container in your `docker-compose` configuration - we'll need to define a simple Dockerfile for it).
+
+Next, we can add Prometheus, AlertManager, and Grafana, a common monitoring stack, to our `docker-compose` configuration. Here is an example configuration that we can adapt: https://dzlab.github.io/monitoring/2021/12/30/monitoring-stack-docker/. AlertManager is used for notifying operators of unexpected conditions, and Grafana is useful for building dashboards that allow us to troubleshoot and understand our system's operation.
+
+If your computer is struggling to run such a complex `docker-compose` system in a performant fashion, you can cut down the number of Kafka topics and consumers that you are running to a minimum (just one producer and consumer/retry consumer pair are fine - don't run sets of these for multiple clusters if your computer is under too much load). 
+
+We'll need to set up a Prometheus configuration to scrape our producers and consumers. Prometheus [configuration](https://prometheus.io/docs/prometheus/latest/configuration/configuration/) is quite complex but we can adapt this [example configuration](https://github.com/prometheus/prometheus/blob/main/documentation/examples/prometheus.yml).
+
+Check that Prometheus is correctly scraping our metrics via http://localhost:9090/metrics.
```

Comment:
I think this could do with slightly more fleshing out - e.g. is the URL saying how to view Prometheus, or what the endpoint Prometheus should be scraping is?